adapters/foo.js

```diff
@@ -0,0 +1,46 @@
+'use strict';
+
+var Thing = require('../thing.js').Thing;
+
+class FooThing extends Thing {
+    constructor(adapter, name) {
+        let properties = {
+            'temperature': {
+                'type': 'integer',
+                'unit': 'celsius',
+                'description': 'An ambient temperature sensor',
+                'value': 23,
+            },
+            'humidity': {
+                'type': 'integer',
+                'unit': 'percent',
+                'value': 35,
+            },
+            'led': {
+                'type': 'boolean',
+                'description': 'A red LED',
+                'value': false,
+            },
+        };
+        super(adapter, 'FooThing', name, properties, [], []);
+    }
+
+    setPropertyValue(name, value) {
+        // This function should propogate the value to the hardware
+        // Calling super.setPropertyValue will update the value cache
+        // and send a 'value-changed' event to any listeners.
+        super.setPropertyValue(name, value);
+    }
+}
+
+class FooAdapter {
```

Comment:
The Thing class is adapter agnostic. FooThing is the specific Thing that needs to be adapter aware. So FooThing and FooAdapter are just dummy classes that I used.

I put them in the same file since they're intimately tied together. For real adapters, they can go in separate files. My intention was to allow the adapters directory to have subdirectories for multi-file adapters.